### PR TITLE
Database integration and better configuration

### PR DESCRIPTION
**Pull request by mmarano25**
Set up database with two tables Tasks and Events for now. 
Made configuration more robust by creating a factory function to create the Flask app and doing configuration there instead of in the main namespace.
Added database access functions in db.py. 
Added database initialization command to the Dockerfile.

### CODE REVIEW

**Comment by mmarano25** [comment]
This is ready to merge.

File: backend/main.py [hunk 1]
```diff
@@ -1,48 +1,68 @@
...
 """
-# TODO: more robust query evaluation
+import os
+
 from flask import Flask, request, jsonify
+
 from recommendation.RecommendationResource import initModel, recommendTasks
 from data_collection.scraper_scheduler import initScheduler
-from data_collection.web_scraper import scrapeARC, get_next_n_events
+from data_collection.web_scraper import scrapeArc, get_next_n_events
```
Comment (mmarano25) [inline backend/main.py]:
```suggestion
from data_collection.web_scraper import scrapeArc, get_next_n_events
```
This should be 
```suggestion
from data_collection.web_scraper import scrapeARC, get_next_n_events
```

File: backend/main.py [hunk 2]
```diff
@@ -1,48 +1,68 @@
...
+    app.config.from_mapping(
+        DATABASE=os.path.join("/home/matt/Desktop/winter_quarter_2019/cs125/project/MESA/backend", "flaskr.sqlite"),
+    )
 
-    Returns:
-        JSON list of floats representing ranked recommendations
+    # Register initialization functions for machine learning and web scraping
+    app.before_first_request(initModel)
+    app.before_first_request(initScheduler)
+    app.before_first_request(scrapeArc)
```
Comment (mmarano25) [inline backend/main.py]:
```suggestion
    app.before_first_request(scrapeARC)
```

File: backend/main.py [hunk 3]
```diff
@@ -1,48 +1,68 @@
...
-def recommendation():
-    """
-    GET route that takes two parameters, hands them off to recommendation
-    algorithm, and sends back the result. 
-    Params:
-        num_resources: int
-        query: [float]
+    # Set database file path
+    app.config.from_mapping(
+        DATABASE=os.path.join("/home/matt/Desktop/winter_quarter_2019/cs125/project/MESA/backend", "flaskr.sqlite"),
```
Comment (mmarano25) [inline backend/main.py]:
```suggestion
        DATABASE="./flaskr.sqlite"),
```

File: backend/main.py [hunk 4]
```diff
@@ -1,48 +1,68 @@
...
-def recommendation():
-    """
-    GET route that takes two parameters, hands them off to recommendation
-    algorithm, and sends back the result. 
-    Params:
-        num_resources: int
-        query: [float]
+    # Set database file path
+    app.config.from_mapping(
+        DATABASE="./flaskr.sqlite"),
```
Comment (kajdreef) [inline backend/main.py]: This line should be changed to: 
`DATABASE="./flaskr.sqlite"`

instead of: `DATABASE="./flaskr.sqlite"),`

Comment (mmarano25) [reply backend/main.py]: Good catch! I modified it on Github when I saw the absolute path, and I must have missed that. I won’t be able to fix it until late tonight, if you want to change it here and commit then merge the pull that’d be awesome. If not I can do it later!

Comment (kajdreef) [reply backend/main.py]: I made the change! :) I will merge it now for you! Great job!!

**Pull request merged**